bin/build_cmd.ml

```diff
@@ -54,6 +54,18 @@ let run_build_system ~common ~request =
             ())
       in
       let* res = run ~toplevel in
+      let* () =
+        match Common.dump_gc_stats common with
+        | None -> Fiber.return ()
+        | Some file ->
+          let path = Path.of_filename_relative_to_initial_cwd file in
+          let+ gc_stat =
+            Gc.full_major ();
+            Gc.compact ();
+            Fiber.return (Gc.stat ())
+          in
+          Dune_util.Gc.serialize ~path gc_stat
+      in
```

Comment:
If instead you use an exit hook, this option will work for every command for free.